src/maa.c

```diff
@@ -0,0 +1,415 @@
+/*
+ * maa.c - Memory Access Amortization (MAA) Implementation
+ *
```

Comment:
As discussion before, please drop all of the references to "memory access amortization". We can call this function either "command_batching" or "memory_prefetching" if we intend to use this more generally for other prefetching.

Reply:
Done.